agent/consul/server.go

```diff
@@ -651,18 +661,21 @@ func NewServer(config *Config, flat Deps, publicGRPCServer *grpc.Server) (*Serve
 	return s, nil
 }
 
-func newFSMFromConfig(logger hclog.Logger, gc *state.TombstoneGC, config *Config) *fsm.FSM {
-	deps := fsm.Deps{Logger: logger}
+func newFSMFromConfig(logger hclog.Logger, gc *state.TombstoneGC, config *Config, publisher *stream.EventPublisher) *fsm.FSM {
+	deps := fsm.Deps{
+		Logger: logger,
+		NewStateStore: func() *state.Store {
+			return state.NewStateStore(gc)
+		},
+	}
+
 	if config.RPCConfig.EnableStreaming {
+		deps.Publisher = publisher
```

Comment:
I think it's no longer the case that we only want to set the publisher if streaming is enabled, right? Will `WatchRoots` and `WatchServers` depend on this config flag?

Reply:
This is a good point. I think it should be safe to always pass in an event publisher and thus remove this EnableStreaming check.

The main difference will be that we are actually tracking events whereas before if streaming was disabled we would configure the state store with a NoOpEventPublisher.

I think its sufficient to allow publishing events but have the config just disable the gRPC APIs around streaming.